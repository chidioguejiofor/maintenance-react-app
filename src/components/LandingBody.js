import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';


const LandingBody = ({
  buttons,

}) => (
  <div className="simple-background-image landing-body">
    <div>
      <h1>Solving your maintenance needs is what we do</h1>

      <ul className="horizontal-list">
        {buttons.map(button => (
          <li key={button.caption}>

            <Link
              className={button.className}
              to={button.link}
              activeClassName={button.className}
            >
              {button.caption}
            </Link>

          </li>
        ))}
      </ul>

    </div>

  </div>
);
export default LandingBody;


LandingBody.propTypes = {
  buttons: PropTypes.arrayOf(PropTypes.object),
};


LandingBody.defaultProps = {
  buttons: []
};
