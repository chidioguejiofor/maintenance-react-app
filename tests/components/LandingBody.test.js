/* eslint no-undef:off */
import React from 'react';
import { shallow } from 'enzyme';

import LandingBody from '../../src/components/LandingBody';

const buttons = [{
  className: 'temp',
  link: 'badlink.com'
}];
describe('Testing LandingBody Component', () => {
  it('should match existing snapshot', () => {
    expect(shallow(<LandingBody buttons={buttons} />)).toMatchSnapshot();
  });
});
