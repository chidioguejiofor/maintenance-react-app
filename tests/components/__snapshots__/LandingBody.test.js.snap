// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing LandingBody Component should match existing snapshot 1`] = `
<div
  className="simple-background-image landing-body"
>
  <div>
    <h1>
      Solving your maintenance needs is what we do
    </h1>
    <ul
      className="horizontal-list"
    >
      <li>
        <Link
          activeClassName="temp"
          className="temp"
          replace={false}
          to="badlink.com"
        />
      </li>
    </ul>
  </div>
</div>
`;
